### PR TITLE
XFCE

### PR DESCRIPTION
**Pull request by triallax**
TODO:
- [x] package core apps (and plugins too? might leave them for later though because there's a lot of them)
- [x] update checks
- [x] recheck dependencies and licenses
- [x] cross-reference templates against other distros to check if i missed anything
- [ ] testing
- [x] generate configure for everything

Broken stuff I noticed so far:
- [x] notifications
- [x] some icons are missing for some reason

### CODE REVIEW

File: contrib/xfdesktop/patches/use-chimera-wallpaper.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/common/xfdesktop-common.h b/common/xfdesktop-common.h
+index 4e1a52ef..65f26bcd 100644
+--- a/common/xfdesktop-common.h
++++ b/common/xfdesktop-common.h
+@@ -35,7 +35,7 @@
+ #include <stdarg.h>
+ 
+ #define XFDESKTOP_CHANNEL        "xfce4-desktop"
+-#define DEFAULT_BACKDROP         DATADIR "/backgrounds/xfce/xfce-shapes.svg"
++#define DEFAULT_BACKDROP         DATADIR "/backgrounds/chimera/bg-l.svg"
```
Comment (triallax) [inline contrib/xfdesktop/patches/use-chimera-wallpaper.patch:10]: should add a dep for this

File: contrib/xfce4-settings/template.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    "libxrandr-devel",
+    "upower-devel",
+    "xfconf-devel",
+    "xorgproto",
+    "xserver-xorg-input-libinput-devel",
+]
+pkgdesc = "Xfce settings app"
+maintainer = "triallax <triallax@tutanota.com>"
+license = "GPL-2.0-only"
+url = "https://xfce.org"
```
Comment (triallax) [inline contrib/xfce4-settings/template.py:43]: might want to use urls to gitlab repos instead?

File: contrib/tumbler/template.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+pkgname = "tumbler"
+pkgver = "4.18.2"
+pkgrel = 0
+build_style = "gnu_configure"
+make_cmd = "gmake"
+hostmakedepends = [
+    "automake",
+    "gettext",
```
Comment (q66) [inline contrib/tumbler/template.py:8]: this doesn't build without this being gettext-devel

Comment (q66) [reply contrib/tumbler/template.py:8]: so far also found in xfce4-settings, possibly others

File: contrib/libkeybinder3/template.py [hunk 4]
```diff
@@ -1,29 +1,26 @@
...
 ]
 makedepends = ["gtk+3-devel"]
 pkgdesc = "Library for registering global keyboard shortcuts (GTK+3)"
 maintainer = "yopito <pierre.bourgin@free.fr>"
 license = "MIT"
 url = "https://github.com/engla/keybinder"
-source = (
-    f"https://github.com/engla/keybinder/archive/keybinder-3.0-v{pkgver}.tar.gz"
-)
+source = f"{url}/archive/{pkgname}-3.0-v{pkgver}.tar.gz"
```
Comment (q66) [inline contrib/libkeybinder3/template.py:22]: 404 link, this change is incorrect

File: contrib/xfce4-screenshooter/template.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
+pkgname = "xfce4-screenshooter"
```
Comment (q66) [inline contrib/xfce4-screenshooter/template.py:1]: this is not depended on by any metapackage

File: contrib/xfce4/template.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
+pkgname = "xfce4"
+pkgver = "4.18"
+pkgrel = 0
+build_style = "meta"
+depends = [
+    "adw-gtk3",
+    "adw-xfwm4",
+    "fonts-dejavu",
+    "gdm",
```
Comment (q66) [inline contrib/xfce4/template.py:9]: we don't want that dependency

File: contrib/xfce4/template.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
+pkgname = "xfce4"
+pkgver = "4.18"
+pkgrel = 0
+build_style = "meta"
+depends = [
+    "adw-gtk3",
+    "adw-xfwm4",
```
Comment (q66) [inline contrib/xfce4/template.py:7]: the theme dependencies should be on the things that declare that the theme is used

File: contrib/xfce4/template.py [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    "fonts-dejavu",
```
Comment (q66) [inline contrib/xfce4/template.py:8]: same with font

File: contrib/xfce4/template.py [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    "gdm",
+    "papirus-icon-theme",
```
Comment (q66) [inline contrib/xfce4/template.py:10]: same with icon theme

File: contrib/tumbler/template.py (hunk 3 above)
Comment (triallax) [reply contrib/tumbler/template.py:8]: strange, i build-tested literally everything but okay

File: contrib/xfce4/template.py (hunk 7 above)
Comment (triallax) [reply contrib/xfce4/template.py:7]: sorry i don't quite get what that means, can you rephrase?

File: contrib/libisofs/template.py [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+pkgrel = 0
+build_style = "gnu_configure"
+hostmakedepends = ["pkgconf", "automake", "libtool"]
+makedepends = ["zlib-devel", "acl-devel"]
+pkgdesc = "Library to create ISO 9660 images"
+maintainer = "triallax <triallax@tutanota.com>"
+license = "GPL-2.0-or-later"
+url = "https://dev.lovelyhq.com/libburnia/libisofs"
+# Remove .pl01 on next bump
+source = f"http://files.libburnia-project.org/releases/{pkgname}-{pkgver}.pl01.tar.gz"
```
Comment (triallax) [inline contrib/libisofs/template.py:12]: might want to add `.1` to `pkgver` instead maybe

File: contrib/xfce4/template.py (hunk 7 above)
Comment (nekopsykose) [reply contrib/xfce4/template.py:7]: the patch for using adw-gtk3 is in xfce4-settings so it should be depended on in that instead of in xfce4

same for the others that do something similar

Comment (triallax) [reply contrib/xfce4/template.py:7]: makes sense yeah

Comment (triallax) [reply contrib/xfce4/template.py:7]: though i don't think there's anything in particular that depends on `fonts-dejavu` so i'm not sure where to put that, or perhaps we can have a `base-desktop-xfce4` which would pull in `base-desktop` which does bring along some font packages?

File: contrib/xfce4-panel/template.py [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+maintainer = "triallax <triallax@tutanota.com>"
+license = "GPL-2.0-or-later AND LGPL-2.1-or-later"
+url = "https://xfce.org"
+source = f"$(XFCE_SITE)/xfce/{pkgname}/{pkgver[:-2]}/{pkgname}-{pkgver}.tar.bz2"
+sha256 = "21337161f58bb9b6e42760cb6883bc79beea27882aa6272b61f0e09d750d7c62"
+# TODO
+options = ["!check"]
+
+
+@subpackage("xfce4-panel-devel")
```
Comment (triallax) [inline contrib/xfce4-panel/template.py:39]: maybe this should be renamed `libxfce4panel-devel` but not sure

File: contrib/adw-xfwm4/template.py [hunk 12]
```diff
@@ -0,0 +1,15 @@
+pkgname = "adw-xfwm4"
+pkgver = "0.0.1"
+_commit = "b0b163bac7d74e5c2d69451d9b1315389bb3c361"
+pkgrel = 0
+pkgdesc = "Libadwaita theme for xfwm4"
+maintainer = "triallax <triallax@tutanota.com>"
+license = "LGPL-2.1-or-later"
```
[the remark this answers is not in the document]
Comment (q66) [reply contrib/adw-xfwm4/template.py:7]: is -later

File: contrib/catfish/template.py [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+hostmakedepends = [
+    "intltool",
+    "python-build",
+    "python-distutils-extra",
+    "python-installer",
+    "python-setuptools",
+    "python-wheel",
+]
+depends = [
+    "gdk-pixbuf",
```
Comment (q66) [inline contrib/catfish/template.py:14]: you can drop gdk-pixbuf, glib, pango, python

File: contrib/exo/template.py [hunk 14]
```diff
@@ -3,23 +3,26 @@
...
     "intltool",
-    "gettext",
-    "glib-devel",
+    "libtool",
+    "pkgconf",
     "python",
+    "xfce4-dev-tools",
 ]
-makedepends = ["glib-devel", "libxfce4ui-devel", "gtk+3-devel"]
+makedepends = ["gtk+3-devel", "glib-devel", "libxfce4ui-devel"]
```
Comment (q66) [inline contrib/exo/template.py:19]: incorrect ordering here

File: contrib/catfish/template.py [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+    "python-pexpect",
+    "virtual:cmd:locate!chimerautils-extra",
+    "xfconf",
+]
+pkgdesc = "Xfce file search tool"
+maintainer = "triallax <triallax@tutanota.com>"
+# TODO: https://gitlab.xfce.org/apps/catfish/-/issues/106
+license = "GPL-2.0-or-later"
+url = "https://xfce.org"
+source = f"$(XFCE_SITE)/apps/{pkgname}/{pkgver[:-2]}/{pkgname}-{pkgver}.tar.bz2"
```
Comment (q66) [inline contrib/catfish/template.py:30]: we generally avoid expanding pkgname in source url these days (ditto everywhere else)

File: contrib/xfburn/template.py [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+    "glib-devel",
+    "gst-plugins-base-devel",
+    "gstreamer-devel",
+    "gtk+3-devel",
+    "libburn-devel",
+    "libgudev-devel",
+    "libisofs-devel",
+    "libxfce4ui-devel",
+]
+depends = ["gst-plugins-bad", "gst-plugins-base", "gst-plugins-good"]
```
Comment (q66) [inline contrib/xfburn/template.py:26]: i don't think the explicit plugin dependencies are needed?

File: contrib/xfce4-mixer/template.py [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+    "gettext-devel",
+    "glib-devel",
+    "gmake",
+    "intltool",
+    "libtool",
+    "pkgconf",
+    "xfce4-dev-tools",
+]
+makedepends = [
+    "alsa-lib-devel",
```
Comment (q66) [inline contrib/xfce4-mixer/template.py:17]: we are generally not interested in enabling alsa backends in stuff that supports pulse/pw

File: contrib/xfce4-panel/template.py [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+    "garcon-devel",
+    "glib-devel",
+    "gtk+3-devel",
+    "libdbusmenu-devel",
+    "libwnck-devel",
+    "libxfce4ui-devel",
+    "libxfce4util-devel",
+    "xfconf-devel",
+]
+pkgdesc = "Xfce settings app"
```
Comment (q66) [inline contrib/xfce4-panel/template.py:29]: this is not a settings app

File: contrib/xfce4-screenshooter/template.py [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+maintainer = "triallax <triallax@tutanota.com>"
+license = "GPL-2.0-or-later"
+url = "https://xfce.org"
+source = f"$(XFCE_SITE)/apps/{pkgname}/{'.'.join(pkgver.split('.')[:-1])}/{pkgname}-{pkgver}.tar.bz2"
+sha256 = "fa711f2a6a5517f575f2e129fe48c2678e836bd4ede5433075f3076d7670621c"
+# TODO
+options = ["!check"]
+
+
+@subpackage(f"{pkgname}-imgur")
```
Comment (q66) [inline contrib/xfce4-screenshooter/template.py:42]: no {pkgname}

File: contrib/xfce4-settings/template.py [hunk 20]
```diff
@@ -0,0 +1,47 @@
...
+    "libxi-devel",
+    "libxklavier-devel",
+    "libxrandr-devel",
+    "upower-devel",
+    "xfconf-devel",
+    "xorgproto",
+    "xserver-xorg-input-libinput-devel",
+]
+# See patches/defaults.patch
+depends = ["adw-gtk3", "papirus-icon-theme"]
```
Comment (q66) [inline contrib/xfce4-settings/template.py:41]: i'd make these softdeps via a subpackage

File: contrib/xfburn/template.py (hunk 16 above)
Comment (triallax) [reply contrib/xfburn/template.py:26]: not strictly, no

File: contrib/xfce4-settings/template.py (hunk 20 above)
Comment (triallax) [reply contrib/xfce4-settings/template.py:41]: what would i call the subpackage?

File: contrib/xfce4/template.py (hunk 8 above)
Comment (triallax) [reply contrib/xfce4/template.py:6]: from xfce4-settings's xsettings.xml:
```xml
    <property name="FontName" type="string" value="Sans 10"/>
    <property name="MonospaceFontName" type="string" value="Monospace 10"/>
```
given the lack of an explicit font family choice, we could sensibly remove this entirely, and people would either install an appropriate font package manually or if they have `base-desktop` they would get `fonts-dejavu` for free

File: contrib/xfce4-settings/template.py (hunk 20 above)
Comment (triallax) [reply contrib/xfce4-settings/template.py]: i went with `-themes` for both `xfwm4` and `xfce4-settings`, though it sounds a bit awkward

File: contrib/xfce4-settings/template.py [hunk 21]
```diff
@@ -0,0 +1,56 @@
...
+]
+pkgdesc = "Xfce settings app"
+maintainer = "triallax <triallax@tutanota.com>"
+license = "GPL-2.0-only"
+url = "https://docs.xfce.org/xfce/xfce4-settings/start"
+source = f"$(XFCE_SITE)/xfce/xfce4-settings/{pkgver[:-2]}/xfce4-settings-{pkgver}.tar.bz2"
+sha256 = "f10c55d0360308d9944f415645d9596d4352f952a20fc7c4a66f30fe511ca1dc"
+
+
+@subpackage("xfce4-settings-themes")
```
Comment (q66) [inline contrib/xfce4-settings/template.py:48]: i'd actually call this `xfce4-settings-theme-defaults` for less confusing name i guess

File: contrib/xfdesktop/template.py [hunk 22]
```diff
@@ -0,0 +1,48 @@
...
+license = "GPL-2.0-or-later"
+url = "https://docs.xfce.org/xfce/xfdesktop/start"
+source = f"$(XFCE_SITE)/xfce/xfdesktop/{pkgver[:-2]}/xfdesktop-{pkgver}.tar.bz2"
+sha256 = "ef9268190c25877e22a9ff5aa31cc8ede120239cb0dfca080c174e7eed4ff756"
+# TODO
+options = ["!check"]
+
+
+@subpackage("xfce4-backgrounds")
+def _backgrounds(self):
```
Comment (q66) [inline contrib/xfdesktop/template.py:43]: i'd make this install_if on the main package

File: contrib/xfce4-settings/template.py (hunk 21 above)
Comment (q66) [reply contrib/xfce4-settings/template.py:48]: or actually maybe `xfce4-settings-default-themes`

File: contrib/xfwm4/template.py [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+]
+pkgdesc = "Xfce window manager"
+maintainer = "triallax <triallax@tutanota.com>"
+license = "GPL-2.0-or-later"
+url = "https://docs.xfce.org/xfce/xfwm4/start"
+source = f"$(XFCE_SITE)/xfce/xfwm4/{pkgver[:-2]}/xfwm4-{pkgver}.tar.bz2"
+sha256 = "92cd1b889bb25cb4bc06c1c6736c238d96e79c1e706b9f77fad0a89d6e5fc13f"
+
+
+@subpackage("xfwm4-themes")
```
Comment (q66) [inline contrib/xfwm4/template.py:39]: ditto here; i'd call it `xfwm4-default-themes`

File: contrib/xfwm4/template.py [hunk 24]
```diff
@@ -0,0 +1,47 @@
...
+def _themes(self):
+    self.pkgdesc = f"{pkgdesc} - default themes"
```
Comment (q66) [inline contrib/xfwm4/template.py:41]: we don't use dashes as the subpackage convention

File: contrib/xfce4-settings/template.py [hunk 25]
```diff
@@ -0,0 +1,56 @@
...
+def _themes(self):
+    self.pkgdesc = f"{pkgdesc} - default themes"
```
Comment (q66) [inline contrib/xfce4-settings/template.py:50]: ditto here with the dash
